### PR TITLE
fix for the spec

### LINKED ISSUE #12: fix the types for the models

### PR DESCRIPTION
**Pull request by baherami**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -333,4 +778,235 @@ components:
...
+      description: Portfolio object
+      properties:
+        id:
+          type: string
+          format: uuid
+          readOnly: true
+        totalAmount:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Total amount in cents
```
Comment (anttileppa) [inline swagger.yaml]:
```suggestion
          description: Total amount in euros
```

File: swagger.yaml [hunk 2]
```diff
@@ -333,4 +778,235 @@ components:
...
+          readOnly: true
+        marketValueTotal:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Market total amount in cents
```
Comment (anttileppa) [inline swagger.yaml]:
```suggestion
          description: Market total amount in euros
```

File: swagger.yaml [hunk 3]
```diff
@@ -333,4 +778,235 @@ components:
...
+          readOnly: true
+        purchaseTotal:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Purchase total amount in cents
```
Comment (anttileppa) [inline swagger.yaml]:
```suggestion
          description: Purchase total amount in euros
```

File: swagger.yaml [hunk 4]
```diff
@@ -333,4 +778,235 @@ components:
...
+      type: object
+      description: Portfolio statistics object
+      required:
+        - subscriptions
+        - redemptions
+      properties:
+        subscriptions:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Subscriptions in cents
```
Comment (anttileppa) [inline swagger.yaml]:
```suggestion
          description: Subscriptions in euros
```

File: swagger.yaml [hunk 5]
```diff
@@ -333,4 +778,235 @@ components:
...
+          readOnly: true
+        redemptions:
+          type: string
+          pattern: /^[+-]?([0-9]+\.?[0-9]*|\.[0-9]+)$/
+          description: Redemptions in cents
```
Comment (anttileppa) [inline swagger.yaml]:
```suggestion
          description: Redemptions in euros
```

File: swagger.yaml [hunk 6]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: string
+          format: uuid
+          description: Fund ID. Can be used to search fund details.
+          readOnly: true
+        amount:
+          type: number
+          format: double
+          description: Amount of funds
+          readOnly: true
+        purchaseValue:
```
Comment (anttileppa) [inline swagger.yaml:855]: Should this field be string?

File: swagger.yaml [hunk 7]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: integer
+          format: int64
+          description: Purchase value in cents
+          readOnly: true
+        totalValue:
```
Comment (anttileppa) [inline swagger.yaml:860]: Should this field be string?

File: swagger.yaml [hunk 8]
```diff
@@ -333,4 +778,235 @@ components:
...
+          description: Target fund ID. Only available when transaction type is security
+          readOnly: true
+        transactionType:
+          $ref: "#/components/schemas/TransactionType"
+        valueDate:
+          type: string
+          format: date
+          description: Date of fund value
+          readOnly: true
+        value:
```
Comment (anttileppa) [inline swagger.yaml:902]: Should this field be string?

File: swagger.yaml [hunk 9]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: integer
+          format: int64
+          description: Fund value in cents
+          readOnly: true
+        shareAmount:
```
Comment (anttileppa) [inline swagger.yaml:907]: Should this field be string?

File: swagger.yaml [hunk 10]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: number
+          format: double
+          description: Total share amount
+          readOnly: true
+        marketValue:
```
Comment (anttileppa) [inline swagger.yaml:912]: Should this field be string?

File: swagger.yaml [hunk 11]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: integer
+          format: int64
+          description: Market value in cents
+          readOnly: true
+        totalValue:
```
Comment (anttileppa) [inline swagger.yaml:917]: Should this field be string?

File: swagger.yaml [hunk 12]
```diff
@@ -333,4 +778,235 @@ components:
...
+          type: integer
+          format: int64
+          description: Total value in cents
+          readOnly: true
+        paymentDate:
+          type: string
+          format: date
+          description: Date when transaction is completed
+          readOnly: true
+        provision:
```
Comment (anttileppa) [inline swagger.yaml:927]: Should this field be string?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: swagger.yaml (hunk 8 above)
Comment (baherami) [reply swagger.yaml:902]: maybe, I havent reached to this part of the implementation. I check.

File: swagger.yaml (hunk 9 above)
Comment (baherami) [reply swagger.yaml:907]: yes same as commented earlier

File: swagger.yaml (hunk 11 above)
Comment (baherami) [reply swagger.yaml:917]: i will check

File: swagger.yaml (hunk 12 above)
Comment (baherami) [reply swagger.yaml:927]: i will check.

**Pull request merged**